neqo-http3/src/recv_message.rs

```diff
@@ -323,10 +323,27 @@ impl RecvStream for RecvMessage {
         matches!(self.state, RecvMessageState::Closed)
     }
 
-    fn stream_reset(&self, app_error: AppError) {
+    fn stream_reset_recv(&self, app_error: AppError, decoder: &mut QPackDecoder) {
+        if !matches!(
+            self.state,
+            RecvMessageState::ClosePending | RecvMessageState::Closed
+        ) || !self.blocked_push_promise.is_empty()
+        {
+            decoder.cancel_stream(self.stream_id);
+        }
         self.conn_events.reset(self.stream_id, app_error);
     }
 
+    fn stream_reset(&self, decoder: &mut QPackDecoder) {
+        if !matches!(
+            self.state,
+            RecvMessageState::ClosePending | RecvMessageState::Closed
```

Comment:
This `matches` is used a few times, would putting it in a little helper function named `closed()`(?) on `RecvMessageState` be worthwhile?